etl/steps/data/meadow/wb/2023-04-30/wb_income.py

```diff
@@ -0,0 +1,90 @@
+"""Load a snapshot and create a meadow dataset."""
+
+from typing import List
+
+import numpy as np
+import pandas as pd
+from owid.catalog import Table
+from structlog import get_logger
+
+from etl.helpers import PathFinder, create_dataset
+from etl.snapshot import Snapshot
+
+# Initialize logger.
+log = get_logger()
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+
+def run(dest_dir: str) -> None:
+    log.info("wb_income: start")
+
+    #
+    # Load inputs.
+    #
+    # Retrieve snapshot.
+    snap: Snapshot = paths.load_dependency("wb_income.xlsx")
+
+    # Load data from snapshot.
+    dfs = pd.read_excel(snap.path, sheet_name=None)
+
+    #
+    # Process data.
+    #
+    # Load sheet with historical WB classifications
+    df = dfs["Country Analytical History"]
+    # Minor formatting (only keep cells with relevant data)
+    df = extract_data_from_excel(df)
+    # Pivot to have years as rows
+    # We could leave this for Garden, but catalog.Table won't accept columns starting with a number. We could change these
+    # to be 2020 -> _2020, but it feels inefficient
+    df = df.melt(id_vars=["country_code", "country"], var_name="year", value_name="classification")
+    # Create a new table and ensure all columns are snake-case.
+    tb = Table(df, short_name=paths.short_name, underscore=True)
+
+    #
+    # Save outputs.
+    #
+    # Create a new meadow dataset with the same metadata as the snapshot.
+    ds_meadow = create_dataset(dest_dir, tables=[tb], default_metadata=snap.metadata)
+
+    # Save changes in the new garden dataset.
+    ds_meadow.save()
+
+    log.info("wb_income: end")
+
+
+def extract_data_from_excel(df: pd.DataFrame) -> pd.DataFrame:
+    # Sanity shape check
+    log.info("wb_income: sanity check on table dimensions")
+    assert (shape := df.shape) == (
+        shape_expected := (239, 37)
```

Comment:
Is this a good thing to assert? On the next update it will definitely change. Maybe it's better to assert that the shape is = 239 >= 37.